src/compose/utils.ts

```diff
@@ -230,7 +230,8 @@ export function getHealthcheck(
 		composeHealthcheck,
 	);
 
-	return _.defaults(composeServiceHealthcheck, imageServiceHealthcheck);
+	// Overlay any compose healthcheck fields on the image healthchecks
+	return _.assign(imageServiceHealthcheck, composeServiceHealthcheck);
```

Comment:
This will still overwrite the image's healthcheck if no healthcheck is provided, because the compose healthcheck will have `test: [ 'NONE' ]` if it was empty. I think you should remove the `healthcheck == null` check in line 194, and have this line be:

```
return _.assign({ test: [ 'NONE' ] }, imageServiceHealthcheck, composeServiceHealthcheck);
```